packages/compartment-mapper/src/archive.js

```diff
@@ -19,20 +19,22 @@ import { compartmentMapForNodeModules } from './node-modules.js';
 import { search } from './search.js';
 import { link } from './link.js';
 import { makeImportHookMaker } from './import-hook.js';
-import { parseJson } from './parse-json.js';
-import { parseArchiveCjs } from './parse-archive-cjs.js';
-import { parseArchiveMjs } from './parse-archive-mjs.js';
+import parserJson from './parse-json.js';
+import parserArchiveCjs from './parse-archive-cjs.js';
+import parserArchiveMjs from './parse-archive-mjs.js';
 import { parseLocatedJson } from './json.js';
 import { unpackReadPowers } from './powers.js';
 import { assertCompartmentMap } from './compartment-map.js';
 
 const textEncoder = new TextEncoder();
 
-/** @type {Record<string, ParseFn>} */
+/** @type {Record<string, ParserImplementation>} */
 const parserForLanguage = {
-  mjs: parseArchiveMjs,
-  cjs: parseArchiveCjs,
-  json: parseJson,
+  mjs: parserArchiveMjs,
+  'pre-mjs-json': parserArchiveMjs,
```

Comment:
`Language` type is used to mean 2 things. One of them is the whole set - possible values for parser, the other is just cjs,mjs,json - the list of possible file types/languages.

I needed to cover all values in the mapping here. Otherwise I'd need two distinct types and a one way converter from larger type to the limited one.